### ISSUE TITLE
Using `propName:from="data" doesn't work

### ISSUE THREAD

**Issue by green3g** [canjs/can-stache-bindings#396]
I forgot to follow up on this issue from a gitter chat awhile back. [Conversation here](https://gitter.im/canjs/canjs?at=5a4510bbedd2230811e9040e)

It seems that `data` is a "reserved" canjs internal word when passing it to components in a template. 
The object received by the component is different. It contains a `fn` and a `metadata` property. 

https://jsbin.com/huzataxare/edit?html,js,output

**Comment by justinbmeyer** [comment]
We should be able to fix this tomorrow.  Thanks for reporting.

**Comment by justinbmeyer** [comment]
So this is a problem addressed in CanJS 4.0 and might be difficult/impossible to fix in 3.0.  However, a work around is to use `./` like:

```
<home-page object:from="./data" />
```

The problem is that helpers are highest priority in `can-stache` (actually in `can-scope`).  The legacy `data` helper is what is being returned.  

In CanJS 4.0, helpers have been moved to the lowest priority.  This means that to get helpers that conflict with the context, you need to do: `{{#scope.helpers.each( ...)}}`.

I'm closing because the work-around should work for you, and this is fixed in 4.0.

**Comment by justinbmeyer** [comment]
The test I created:

```js
QUnit.test("data should not be a reserved word (#396)", function(){

	var appViewModel = new DefineMap({
		data: 'test'
	});

	var homeVM = new DefineMap({
		page: 'string',
		object: 'string'
	});
	debugger;
	MockComponent.extend({
		tag: "home-page",
		viewModel: homeVM
	});

	var view = stache('<home-page object:from="data" />');
	view(appViewModel);

	QUnit.equal(homeVM.object, "test", "got data passed");

});
```
